code/game/objects/effects/bump_teleporter.dm

```diff
@@ -19,19 +23,51 @@
 	LAZYREMOVE(AllTeleporters, src)
 	return ..()
 
-
 /obj/effect/bump_teleporter/singularity_act()
 	return
 
 /obj/effect/bump_teleporter/singularity_pull()
 	return
 
-/obj/effect/bump_teleporter/Bumped(atom/movable/AM)
-	if(!ismob(AM))
+/obj/effect/bump_teleporter/Bumped(atom/movable/concerned_party)
+	if(!validate_setup(concerned_party))
 		return
+
+	for(var/obj/effect/bump_teleporter/teleporter in AllTeleporters)
+		if(teleporter.id == src.id_target)
+			concerned_party.forceMove(teleporter.loc)
```

Comment:
Instead of doing this, replace this with a `teleport(teleporter.loc)` proc that the do_teleport subtype overrides. That way we can avoid the copy-paste.